File: JAVA API/crops/src/main/java/com/eagroservices/crops/services/CropService.java
package com.eagroservices.crops.services;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.eagroservices.crops.dao.ICropRepository;
import com.eagroservices.crops.pojos.Crop;
import com.eagroservices.crops.pojos.CropNameIdDetails;

@Service
public class CropService implements ICropService {
    @Autowired
    private ICropRepository _repo;

    @Override
    public List<Crop> GetAll() {
        return _repo.findAll();
    }

    @Override
    public Optional<Crop> GetById(int cropId) {
        return _repo.findById(cropId);
    }

    @Override
    public Boolean Insert(Crop crop) {
        return _repo.save(crop) != null;
    }

    @Override
    public Boolean Update(Crop crop) {
        return _repo.save(crop) != null;
    }

    @Override
    public Boolean Delete(int cropId) {
       boolean status= false;
        if (_repo.existsById(cropId)) {
            _repo.deleteById(cropId);
            status=true;
        }
        return status;
    }

    @Override
    public List<String> GetCropNames() {
        return _repo.GetCropNames();
    }

    @Override
    public List<CropNameIdDetails> GetCropNamesWithId() {
        return _repo.GetCropNamesWithId();
    }

}

File: API/Crops/Services/CropService.cs
using Transflower.EAgroServices.Crops.Models;
using Transflower.EAgroServices.Crops.Repositories.Interfaces;
using Transflower.EAgroServices.Crops.Entities;
using Transflower.EAgroServices.Crops.Services.Interfaces;

namespace Transflower.EAgroServices.Crops.Services;

public class CropService : ICropService
{
    private readonly ICropRepository _repository;

    public CropService(ICropRepository repository)
    {
        _repository = repository;
    }

    public async Task<List<Crop>> GetAll() => await _repository.GetAll();

    public async Task<List<string>> GetCropNames() => await _repository.GetCropNames();

    public async Task<List<CropDetail>> GetCropNamesWithId() =>
        await _repository.GetCropNamesWithId();

    public async Task<Crop?> GetById(int cropId) => await _repository.GetById(cropId);

    public async Task<bool> Insert(Crop crop) => await _repository.Insert(crop);

    public async Task<bool> Update(int cropId, Crop crop) => await _repository.Update(cropId, crop);

    public async Task<bool> Delete(int cropId) => await _repository.Delete(cropId);
}
